asab/web/tenant/utils.py

```diff
@@ -37,12 +37,20 @@ def set_handler_tenant(tenant_service, route: aiohttp.web.AbstractRoute):
 
 def _pass_tenant(tenant_service, handler):
 	"""
-	Add tenant to the handler arguments
+	Pass tenant from Tenant context variable to web handler as an argument.
 	"""
 	@functools.wraps(handler)
 	async def wrapper(*args, **kwargs):
 		tenant = Tenant.get()
-		if not tenant_service.is_tenant_known(tenant):
+
+		if tenant is None:
+			if not (hasattr(handler, "AllowNoTenant") and handler.AllowNoTenant is True):
+				raise aiohttp.web.HTTPNotFound(reason="Tenant not found.")
+			else:
+				# `None` is allowed: Tenant is optional at this endpoint.
+				pass
+
+		elif not tenant_service.is_tenant_known(tenant):
 			L.warning("Tenant not found.", struct_data={"tenant": tenant})
 			raise aiohttp.web.HTTPNotFound(reason="Tenant not found.")
```

Comment:
@ateska I think this check you introduced is not necessary here. Tenant existence has already been verified when extracting from URL path or query and setting the Tenant context variable (see the `_set_tenant_context_from_url_query` and `_set_tenant_context_from_url_path` methods below). 
The purpose of this wrapper is to simply pass the tenant context value into handler args.

Reply:
Ok, it was not working properly in SeaCat PKI